napari/components/_layer_slicer.py

```diff
@@ -0,0 +1,150 @@
+from __future__ import annotations
+
+import logging
+from concurrent.futures import Executor, Future, ThreadPoolExecutor
+from threading import RLock
+from typing import Dict, Iterable, Optional, Tuple
+
+from napari.components import Dims
+from napari.layers import Layer
+from napari.utils.events.event import EmitterGroup, Event
+
+logger = logging.getLogger("napari.components._layer_slicer")
+
+
+class _LayerSlicer:
+    """
+    High level class to control the creation of a slice (via a slice request),
+    submit it (synchronously or asynchronously) to a thread pool, and emit the
+    results when complete.
+
+    Attributes
+    ----------
+    events : napari.utils.events.event.EmitterGroup
```

Comment:
You can use an `Events` section as in `SelectableEventableList` and document the `ready` event. Also, though we originally vendored evens from vispy, we probably don't want to reference vispy here.

Reply:
What should the type of the task.result() be? Will that be a LayerResponse? Or, can I leave it generic for now?

Reply:
Left it as `T` similar to other items in the `SelectableEventedList` example you mentioned